torch/nn/modules/module.py

```diff
@@ -357,6 +388,38 @@ def modules(self, memo=None):
                 for m in module.modules(memo):
                     yield m
 
+    def named_modules(self, memo=None, prefix=None):
+        """Returns an iterator over all modules in the network, yielding
+        both the name of the module as well as the module itself.
+
+        Note:
+            Duplicate modules are returned only once. In the following
+            example, ``l`` will be returned only once.
+
+            >>> l = nn.Linear(2, 2)
+            >>> net = nn.Sequential(l, l)
+            >>> for idx, m in enumerate(net.named_modules()):
+            >>>     print(idx, '->', m)
+            0 -> ('', Sequential (
+              (0): Linear (2 -> 2)
+              (1): Linear (2 -> 2)
+            ))
+            1 -> ('0', Linear (2 -> 2))
+        """
+
+        if memo is None:
+            memo = set()
+            prefix = []
+        if self not in memo:
+            memo.add(self)
+            yield '.'.join(prefix), self
```

Comment:
I think it might be faster to use string concatenation instead of joining the lists multiple times in the tree leaves. You can replace `prefix.append` with `submodule_prefix = prefix + '.' + name` at the beginning of the loop.